APMrover2/mode_breadcrumb.cpp

```diff
@@ -0,0 +1,183 @@
+#include "mode.h"
+#include "Rover.h"
+
+
+bool ModeBreadcrumb::_enter()
+{
+    //don't call enter again if _crumbs exists
+    if (_crumbs != nullptr) {
+        return false;
+    }
+
+    if (_crumbs_sem == nullptr){
+        // create semaphore
+        _crumbs_sem = hal.util->new_semaphore();
+        if (_crumbs_sem == nullptr) {
+            return false;
+        }
+    }
+
+    _current_crumb = nullptr;
+
+    // allocate crumbs queue
+    _crumbs = (Location*)calloc(crumbs_max, sizeof(Location));
```

Comment:
Return value should be checked here.  We'll call guided's enter even if this allocation fails.

At this point it might not be a bad idea to have a `goto FAIL:`... but you'll need to make that call after making the changes...

Reply:
So for the `goto FAIL`, it should called guided's enter, and then return false regardless of the return value of guided's enter?